mu-plugins/blocks/latest-news/latest-news.php

```diff
@@ -0,0 +1,123 @@
+<?php
+/**
+ * Block Name: Latest News
+ * Description: A block for use across the whole wp.org network.
+ *
+ * @package wporg
+ */
+
+namespace WordPressdotorg\MU_Plugins\Latest_News;
+
+/**
+ * Determines whether we can switch blogs.
+ *
+ * @return boolean
+ */
+function should_switch_to_blog() {
+	return function_exists( 'is_multisite' ) && is_multisite();
+}
+
+/**
+ * Renders the `wporg/latest-news` block on server.
+ *
+ * @param array $attributes The block attributes.
+ *
+ * @return string Returns the block content with received post items.
+ */
+function render_block( $attributes ) {
+	$defaults = array(
+		'perPage' => 3,
+	);
+	$attributes = wp_parse_args( $attributes, $defaults );
+	$blog_switched = false;
+
+	// Check if we can switch to the News blog.
+	// @todo Prevent switch if Rosetta, Rosetta should use local posts.
+	if ( should_switch_to_blog() && defined( 'WPORG_NEWS_BLOGID' ) ) {
+		switch_to_blog( WPORG_NEWS_BLOGID );
+		$blog_switched = true;
+	}
+
+	$cache_key = 'wporg-latest-news-' . $attributes['perPage'];
+	$posts = get_transient( $cache_key );
+	if ( ! $posts ) {
+		$posts = wp_get_recent_posts(
+			array(
+				'numberposts' => $attributes['perPage'],
+				'post_status' => 'publish',
+			)
+		);
+
+		if ( is_wp_error( $posts ) ) {
+			return $posts->get_error_message();
+		}
+
+		if ( empty( $posts ) ) {
+			return __( 'No posts found.', 'wporg' );
+		}
```

Comment:
I moved these into the `if` here so that we don't cache errors.

Reply:
👍